sync3/handler/connstate.go

```diff
@@ -93,32 +93,42 @@ func (s *ConnState) load(ctx context.Context) error {
 	for _, metadata := range joinedRooms {
 		metadata.RemoveHero(s.userID)
 		urd := s.userCache.LoadRoomData(metadata.RoomID)
-		rooms[i] = sync3.RoomConnMetadata{
-			RoomMetadata: *metadata,
-			UserRoomData: urd,
+		timestamps := make(map[string]uint64, len(req.Lists))
+		for listKey, _ := range req.Lists {
 			// Best-effort only: we're not going to scan the database for all events in
 			// the entire room's history to give you a fully accurate timestamp
 			// according to your bump_event_types.
-			LastInterestedEventTimestamp: metadata.LastMessageTimestamp,
+			timestamps[listKey] = metadata.LastMessageTimestamp
+		}
```

Comment:
This may not be unnecessary with the fallback logic I added for use in the sorter. But I think it's easier to be explicit as early as you can wherever possible.